src/metabase/driver/sql_jdbc/sync/describe_table.clj

```diff
@@ -131,21 +131,32 @@
                  nil)
    (fn [^ResultSet rs]
       ;; https://docs.oracle.com/javase/7/docs/api/java/sql/DatabaseMetaData.html#getColumns(java.lang.String,%20java.lang.String,%20java.lang.String,%20java.lang.String)
-     #(let [default            (.getString rs "COLUMN_DEF")
+     #(let [;; COLUMN_DEF = "" observed with clickhouse, "" is never a valid SQL expression, so treat as undefined
+            default            (not-empty (.getString rs "COLUMN_DEF"))
             no-default?        (contains? #{nil "NULL" "null"} default)
-            nullable           (.getInt rs "NULLABLE")
-            not-nullable?      (= 0 nullable)
-             ;; IS_AUTOINCREMENT could return nil
+            ;; leave room for "", or other strings to be nil (unknown)
+            is-nullable        ({"YES" true, "NO" false} (.getString rs "IS_NULLABLE"))
+            is-generated       (when (driver/database-supports? driver :describe-is-generated nil) ; if not supported, we might get an inaccurate answer from IS_GENERATEDCOLUMN (e.g clickhouse)
```

Comment:
Not using `nil` for the `database` argument. I think practically this is fine (as I've added the describe-is-generated feature), but I am unsure whether the general contract is such that 3rd party JDBC drivers might specialise this impl for a given `database` and therefore this might break with a null pointer. 

Though, I suppose if they run the tests then they should fine they _do_ need to accept `nil` for this feature, as they will fail.